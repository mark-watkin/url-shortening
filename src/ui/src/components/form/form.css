.form {
    display: flex;
    flex-direction: row;
    gap: 20px;
}

@media only screen and (max-width: 500px) {
    .form {
        flex-direction: column;
        gap: 20px;
    }
}

.form-input {
    padding: 12px 20px;
    border-style: solid;
    border-width: 1px;
    border-color: #d7d7d7;
    border-radius: 25px;
    -webkit-box-sizing: border-box;
    -moz-box-sizing: border-box;
    box-sizing: border-box;
    font-size: 1rem;
    flex: 1;
}

.form-input:focus {
    outline: none;
    border-color: rgb(110, 110, 110);
}

.form-input-invalid-text {
    font-size: 0.875rem;
    line-height: 1.43;
    color: #d73838
}

.form-input-invalid {
    border-color: #d73838 !important;
}