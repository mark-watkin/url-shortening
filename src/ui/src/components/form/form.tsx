import { useRef } from 'react';
import { Button } from '../button';
import './form.css';

export interface FormProps {
    inputHint: string;
    submitText: string;
    invalidText: string;
    isInvalid: boolean;
    onSubmit: (value: string | undefined) => Promise<any>;
}

export const Form: React.FC<FormProps> = ({inputHint, submitText, invalidText, isInvalid, onSubmit}) =>  {
    const input: React.RefObject<HTMLInputElement> = useRef(null);
    
    return <div>
        <form className='form' onSubmit={async (e) => {
            e.preventDefault(); 
            await onSubmit(input.current?.value);
        }}>
            <input className={`form-input ${isInvalid ? 'form-input-invalid' : ''}`} name="url" placeholder={inputHint} ref={input} />
            <Button text={submitText} />
        </form>
        <div className='form-input-invalid-text'>{isInvalid ? invalidText : ''}</div>
    </div>
}