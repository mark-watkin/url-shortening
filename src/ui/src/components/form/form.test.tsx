import renderer from 'react-test-renderer';
import { Form } from './form';

it('renders correctly when in error', () => {
  const tree = renderer
    .create(<Form 
        inputHint="Enter a url to shorten!" 
        submitText="Submit" 
        invalidText="The url that you have entered is not valid. Dont forget to include the protocol!"
        isInvalid={true}
        onSubmit={() => Promise.resolve(0)}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly when not in error', () => {
    const tree = renderer
      .create(<Form 
          inputHint="Enter a url to shorten!" 
          submitText="Submit" 
          invalidText="The url that you have entered is not valid. Dont forget to include the protocol!"
          isInvalid={false}
          onSubmit={() => Promise.resolve(0)}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
});