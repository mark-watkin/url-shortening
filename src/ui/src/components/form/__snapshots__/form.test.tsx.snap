// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when in error 1`] = `
<div>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <input
      className="form-input form-input-invalid"
      name="url"
      placeholder="Enter a url to shorten!"
    />
    <button
      className="button"
    >
      Submit
    </button>
  </form>
  <div
    className="form-input-invalid-text"
  >
    The url that you have entered is not valid. Dont forget to include the protocol!
  </div>
</div>
`;

exports[`renders correctly when not in error 1`] = `
<div>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <input
      className="form-input "
      name="url"
      placeholder="Enter a url to shorten!"
    />
    <button
      className="button"
    >
      Submit
    </button>
  </form>
  <div
    className="form-input-invalid-text"
  />
</div>
`;
